samples/cellular/modem_shell/src/shell.c

```diff
@@ -254,8 +255,32 @@ int heap_shell(const struct shell *shell, size_t argc, char **argv)
 
 int version_shell(const struct shell *shell, size_t argc, char **argv)
 {
+	int ret;
+	char info_str[MODEM_INFO_MAX_RESPONSE_SIZE + 1];
+
 	mosh_print_version_info();
 
+	ret = modem_info_get_hw_version(info_str, sizeof(info_str));
```

Comment:
So these doesn't work in the boot I guess?
Generally I'd accept these into `mosh_print_version_info()` but the boot would be an issue.

Reply:
It doesn't work currently, because `mosh_print_version_info()` is called before modemlib and modem_info library are initialized. That said, I don't see why the order couldn't be changed, if we want to print this information also at boot time.

Reply:
I've done this change now. `mosh_print_version_info()` uses `printk()`, because we've sometimes had problems using `shell_print()` when the application is starting. Now that the function is called a bit later, this may not be a problem. I did not change the implementation though, because I'm not sure how to verify that it works always.